### PR TITLE
feat: feed end indicator (2)

### PR DESCRIPTION
**Pull request by sevonj**
Second attempt at #369 
image

Still needs to use translations, maybe rephrase it too.

I omitted the scroll to top button as it did not work.

### CODE REVIEW

File: src/lib/components/lemmy/post/feed/VirtualFeed.svelte [hunk 1]
```diff
@@ -274,6 +275,16 @@
...
         </div>
       </div>
+    {:else}
+      <div style="border-top-width: 0">
+        <EndPlaceholder>
+          {#if feedData.community_name}
+            You have reached the end of {feedData.community_name}.
+          {:else}
+            You have reached the end of the feed.
+          {/if}
```
Comment (sevonj) [inline src/lib/components/lemmy/post/feed/VirtualFeed.svelte:285]: These strings are now separate because I already know that trying to work one to fit both would be a pain in some languages.

Comment (Xyphyn) [reply src/lib/components/lemmy/post/feed/VirtualFeed.svelte:285]: You could make a string like this:

```
"endFeed": "You have reached the end of {feed}",
"feed": "The feed"
```

and then in the code:

```js
$t('endFeed', { feed: feedData.community_name ?? $t('feed').toLowercase()  }
```

Comment (sevonj) [reply src/lib/components/lemmy/post/feed/VirtualFeed.svelte:285]: I have to think through the logic so I can make it grammatically correct in the Finnish translation. The language uses cases everywhere, so I have to restructure the sentence to work myself out of the need to change the case of community_name.

Maybe I can put the entire sentence inside a check like `{{feed;the feed:{{feed}}_sentence; default:community_sentence {{feed}};}}`

File: src/lib/i18n/en.json [hunk 2]
```diff
@@ -183,7 +183,8 @@
   "routes": {
     "frontpage": {
       "title": "Frontpage",
-      "footer": "{{users:number}} active {{users; 1:user; default:users;}}"
+      "footer": "{{users:number}} active {{users; 1:user; default:users;}}",
+      "endFeed": "You have reached the end of {{community_name; undefined:the feed.; default:{{community_name}}.}}"
```
Comment (sevonj) [inline src/lib/i18n/en.json:187]: This is the cleanest way I could come up with.

I put it under routes.frontpage since that's where the old footer is too.
image

**Comment by Xyphyn** [comment]
I just added a scroll to top button and made it so it wouldn't overflow with the scroll button.

**Comment by Xyphyn** [comment]
Thanks for your contribution!
